geowave-store/src/main/java/mil/nga/giat/geowave/store/filter/SpatialQueryFilter.java

```diff
@@ -188,7 +193,7 @@ public void fromBinary(
 					fieldId));
 		}
 		buf.get(theRest);
-		queryGeometry = GeometryUtils.geometryFromBinary(geometryBinary);
+		queryGeometry = geometryInterner.intern(GeometryUtils.geometryFromBinary(geometryBinary));
```

Comment:
You win again, before I just googled "JTS Geometry source" to be sure equals()/hashCode() were overridden to do topological equality - it turns out I was looking at version 1.11 source...

It changed since then and you're correct we are using a version that overrides equals and hashcode:
http://sourceforge.net/p/jts-topo-suite/code/321/